New feature lumaris rgb tunable white

RBaragona opened the pull request:
Add support for a new LEAP light device type to enable control of Lumaris RGB + Tunable White.

RBaragona commented:
@mdonoughe or @bdraco , anything you'd like to see in order to approve/merge this?  I tested this with a HWQSX system.

eclair4151 commented:
You'll need to add some tests for sure, start by defining a new device/zone/etc inside the test response jsons:
https://github.com/gurumitts/pylutron-caseta/tree/dev/tests/responses/hwqsx

And then add tests in test_smartbridge to confirm the request and response and formatted as expected
https://github.com/gurumitts/pylutron-caseta/blob/dev/tests/test_smartbridge.py

See test_qsx_set_ketra_color for a similar example.

mdonoughe and bdraco, let me know if you want me to test as well, happy to throw a fake tunable RGB on my QSX system with lutron designer and confirm I can control it via the library.

mdonoughe commented on src/pylutron_caseta/smartbridge.py:
```diff
@@ -426,7 +428,7 @@ async def set_value(
             )
             return
-
+            
```
```suggestion

```

mdonoughe commented on src/pylutron_caseta/__init__.py:
```diff
@@ -15,8 +15,9 @@
-        "WhiteTune",  # Lumaris Tape Light
+        "WhiteTune",  # Lumaris Tunable White Tape Light
         "PowPak0-10V",
+        "ColorTune", # Lumaris RGB + Tunable White Tape Light
```
```suggestion
        "ColorTune",  # Lumaris RGB + Tunable White Tape Light
```

mdonoughe commented on src/pylutron_caseta/smartbridge.py:
```diff
@@ -404,8 +406,8 @@ async def set_value(
-        # Handle Ketra lamps
-        if device.get("type") == "SpectrumTune":
+        # Handle Ketra lamps and Lumaris RGB + Tunable White Tape Light
+        if ((device.get("type") == "SpectrumTune") or (device.get("type") == "ColorTune")):
```
```suggestion
        if device.get("type") in ["SpectrumTune", "ColorTune"]:
```

RBaragona commented on tests/responses/hwqsx/device-list.json:
```diff
@@ -176,6 +176,32 @@
+          }
+        ],
+        "DeviceClass": {
+          "HexadecimalEncoding": "61a0101"
```
@eclair4151 I'm not sure how to find the device class, but I also do not see it used, so do not believe this impacts function/testing.

RBaragona commented:
@mdonoughe I believe I have addressed all of your comments.  Please let me know if anything else is needed?

@eclair4151 , I have added tests as you suggested.

I believe this should be ready to merge.

eclair4151 commented:
Looks good to me. I don't have one of those devices to test, but I assume you tried it out on yours. 
I also just raised a quick PR to add a new tilt device type for wood blinds, so hopefully we can get these both merged in, and then make a .23 release
